tests/networkpolicy_test.go

```diff
@@ -187,19 +188,37 @@ func createVMICirros(virtClient kubecli.KubevirtClient, namespace string, labels
 	return vmi
 }
 
-func assertPingSucceed(fromVmi *v1.VirtualMachineInstance, toIp string) {
+func assertPingSucceed(fromVmi, toVmi *v1.VirtualMachineInstance) {
+	ExpectWithOffset(1, toVmi.Status.Interfaces[0].IPs).NotTo(BeEmpty())
 	ConsistentlyWithOffset(1, func() error {
-		return tests.PingFromVMConsole(fromVmi, toIp)
+		for _, toIp := range toVmi.Status.Interfaces[0].IPs {
+			if err := tests.PingFromVMConsole(fromVmi, toIp); err != nil {
+				return err
+			}
+		}
+		return nil
 	}, 15*time.Second, 1*time.Second).Should(Succeed())
 }
 
-func assertPingFail(fromVmi *v1.VirtualMachineInstance, toIp string) {
+func assertPingFail(fromVmi, toVmi *v1.VirtualMachineInstance) {
+	ExpectWithOffset(1, toVmi.Status.Interfaces[0].IPs).NotTo(BeEmpty())
+
 	EventuallyWithOffset(1, func() error {
-		return tests.PingFromVMConsole(fromVmi, toIp)
+		for _, toIp := range toVmi.Status.Interfaces[0].IPs {
+			if err := tests.PingFromVMConsole(fromVmi, toIp); err == nil {
+				return nil
+			}
+		}
+		return fmt.Errorf("Ping failed on all available ips as expected")
```

Comment:
You just need to return the error, not add here any text. It will never be printed anyway.

Reply:
there isnt error to return, so i need to create one, 
`*new(error)` like this ?
its ugly, i guess there are better ways

Reply:
Done both